sqlx-core/src/pool/inner.rs

```diff
@@ -75,6 +80,12 @@ impl<DB: Database> SharedPool<DB> {
         self.is_closed.load(Ordering::Acquire)
     }
 
+    /// Return the [`Duration`] this pool has spent waiting on a permit for a
+    /// connection.
+    pub(super) fn pool_wait_duration(&self) -> Duration {
+        Duration::from_millis(self.pool_wait_duration_ms.load(Ordering::Relaxed))
+    }
```

Comment:
By always returning a `Duration` to the user, we hide implementation detail of the atomic measured value precision / unit conversions.